src/libraries/Common/src/System/Obsoletions.cs

```diff
@@ -171,5 +171,9 @@ internal static class Obsoletions
 
         internal const string AesGcmTagConstructorMessage = "AesGcm should indicate the required tag size for encryption and decryption. Use a constructor that accepts the tag size.";
         internal const string AesGcmTagConstructorDiagId = "SYSLIB0053";
+
+        internal const string ThreadVolatileReadMessage = "Thread.VolatileRead is obsolete. Use Volatile.Read instead.";
```

Comment:
This needs to be added to https://github.com/dotnet/runtime/blob/main/docs/project/list-of-diagnostics.md